blockchain/chain_test.go

```diff
@@ -964,3 +964,202 @@ func TestIntervalBlockHashes(t *testing.T) {
 		}
 	}
 }
+
+func TestChainTips(t *testing.T) {
+	type testcase struct {
+		name      string
+		chain     *BlockChain
+		chainTips map[chainhash.Hash]ChainTip
+	}
+	tests := []testcase{}
```

Comment:
Typically we'd do the initialization+declaration all in one statement. 

So one way to achieve this would be to have the `chainTips` be a closure that returns the chain tip itself:
```go
type testCase struct {
    chainTipGen func() map[chainhash.Hash]ChainTip
}
```

Then you'd have the current block as part of the slice definition itself. 

We typically don't use scoping blocks like this in the manner of C++ as it can make it harder to read/understand a code fragment.